lib/sort_by_length.rb

```diff
@@ -1,7 +1,30 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n ^ 2)
+# Space complexity: ? O(n) because of splitting the string at the beginning of the method
+
+def swap(arr, index_one, index_two)
+  tmp = arr[index_two]
+  arr[index_two] = arr[index_one]
+  arr[index_one] = tmp
+end
+
+
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  my_sentence = my_sentence.split 
+  
+  j = 0 # outer counter
+  my_sentence.length.times do
+    comparison_word = my_sentence[j]
+    for i in j...(my_sentence.length)
+      if my_sentence[i].length < comparison_word.length
+        puts my_sentence[i]
+        puts comparison_word
+        swap(my_sentence, i, j )
+        comparison_word = my_sentence[j]
+      end
```

Comment:
This is an interesting sort, it's not one of the standard sorts but does sort elements.  It does work however.